database/models/features.db.php

```diff
@@ -22,25 +23,25 @@ function __construct(User $user, Feature $feature)
         $this->featureName = $feature->value;
         $this->user = $user;
     }
+}
 
-    /**
-     * Get all feature from a user
-     * @return bool
-     */
-    static function hasFeature($uid, $feature)
-    {
-        return !!em()->createQuery("SELECT count(f) FROM UserFeature f WHERE f.user = :u AND f.featureName = :feature AND f.enabled = :enabled")
-            ->setParameters(["u" => $uid, "feature" => $feature, "enabled" => true])
-            ->getSingleScalarResult();
-    }
+/* Bit different logic here, feature is authorized if there is a line in ClubFeature table, and then can be enabled at club level. */
+#[Entity, Table(name: 'club_features')]
+class ClubFeature
+{
+    #[Id, Column]
+    public string|null $featureName = null;
 
-    /**
-     * list all feature for one user
-     * @return UserFeature[]
-     */
-    static function list($uid)
+    #[Id, ManyToOne, JoinColumn(name: "club_slug", referencedColumnName: "slug")]
```

Comment:
This join table is weird. I don't think you need it.

Reply:
I had some problems with doctrine, because the ID column of the club entity is not a normal ID but a string (maybe ?). I'll try and look a bit more into it.

Reply:
I think I see where the problem comes from: you don't need an Id here
```suggestion
    #[ManyToOne]
```

Reply:
I checked again and the problem is really that the Club entity does not use a column named "id" as ID, but "slug". So it gives a problem when not explicitally specifying the column name.
![image](https://github.com/user-attachments/assets/46a17ff4-d987-49f9-bb1f-dd23c297fc5d)

Reply:
Got it! I'll resolve